Small fixes

nisarhassan12 opened the pull request:
Fixes #177

jankeromnes commented on src/components/Logos.tsx:
```diff
@@ -16,6 +16,7 @@ const Logos: React.SFC<LogosProps> = ({ logos }) => (
                 top,
                 width
             }}
+            alt="Gitpod Logo"
```
Note: I also previously tried to add this, but it creates a very weird effect where the logos on the website briefly show up as text when you load a page. This doesn't look nice, and also, since these logos are purely decorative, I don't think we need an `alt` attribute here.

This is why I added the `aria-hidden='true'` attribute. Please remove this `alt` attribute again.

jankeromnes commented on src/pages/pricing.tsx:
```diff
@@ -232,13 +232,13 @@ export default class PricingPage extends React.Component<{}, {}> {
                         </h3>
                         <p>
-                            Please <a href="mailto:contact@gitpod.io">contact us</a> and we will inform you as soon as <strong>Gitpod Enterprise</strong> is available.
+                            Please contact us and we will inform you as soon as <strong>Gitpod Enterprise</strong> is available.
```
Thanks for noticing this duplicate link! However, this was on purpose, because someone told us it was a good idea to also make the `contact us` words a link.

Please make this into a link again.

jankeromnes commented on src/resources/icons.tsx:
```diff
@@ -133,14 +133,14 @@ export function videoPlay(aStyle?: React.CSSProperties, onClick?: ()=>void) {
 export function arrowLeft(aStyle?: React.CSSProperties) {
     const style = merge(aStyle || {});
-    return <svg xmlns="http://www.w3.org/2000/svg" width="27.913" height="58.337" viewBox="0 0 27.913 58.337" style={style}>
+    return <svg xmlns="http://www.w3.org/2000/svg" width="27.913" height="58.337" viewBox="0 0 27.913 58.337" style={{...style, filter: 'none'}}>
```
Hm, this does work, but it prevents anyone from ever using a `filter` with this icon. Could you please fix this differently, e.g. by adding a `filter: 'none'` rule to all calls to `arrowLeft({})` instead?

nisarhassan12 replied:
Thanks! Sir very godd catch it would be a lot more simpler and feasible if we were to filter: none from where the arrowLeft is rendered into the view.

jankeromnes commented on src/resources/icons.tsx:
```diff
@@ -133,14 +133,14 @@ export function videoPlay(aStyle?: React.CSSProperties, onClick?: ()=>void) {
 export function arrowRight(aStyle?: React.CSSProperties) {
     const style = merge(aStyle || {});
-    return <svg xmlns="http://www.w3.org/2000/svg" width="27.913" height="58.337" viewBox="0 0 27.913 58.337" style={style}>
+    return <svg xmlns="http://www.w3.org/2000/svg" width="27.913" height="58.337" viewBox="0 0 27.913 58.337" style={{...style, filter: 'none'}}>
```
Same as above.

jankeromnes commented on src/components/Boilerplate.tsx:
```diff
@@ -47,7 +47,7 @@ const Wrapper = styled.div`
 `
 
-export const BoilerPlate = (p: {title: string, maxHeight?: number, children: any}) => <IndexLayout title={p.title}>
+export const BoilerPlate = (p: {title: string, maxHeight?: number, children: any, canonical: string}) => <IndexLayout title={p.title} >
```
Don't you also need to pass this `canonical` attribute to `IndexLayout`?

```suggestion
export const BoilerPlate = (p: {title: string, maxHeight?: number, children: any, canonical: string}) => <IndexLayout title={p.title} canonical>
```

nisarhassan12 replied:
Yes I have to. I didn't realized.

jankeromnes commented on src/templates/doc.tsx:
```diff
@@ -70,17 +72,21 @@ const DocTemplate: React.SFC<DocTemplateProps> = ({ data }) => {
     const editUrl = data.markdownRemark.fileAbsolutePath.replace(/^.*\/src\/docs\//, 'https://gitpod.io/#https://github.com/gitpod-io/website/blob/master/src/docs/');
     return (
-        <IndexLayout>
+        <IndexLayout canonical="https://www.gitopd.io/docs">
```
This should be templated with the slug.

jankeromnes commented on src/templates/doc.tsx:
```diff
@@ -14,21 +14,23 @@ import GatsbyLink from 'gatsby-link';
-    width: 200px;
+    position: fixed;
     min-width: 200px;
-    padding-top: 120px;
```
I'm not sure that I like these changes, they make the scrolling behavior a bit weird:

https://youtu.be/d9frbCrf_ZE

Maybe let's revert the styling changes to the Docs Table of Contents here, and maybe move them to a new PR where we can iterate on getting them right.

jankeromnes commented on src/templates/blog.tsx:
```diff
@@ -89,7 +91,7 @@ const TeaserImage = styled.div`
 const BlogTemplate: React.SFC<BlogTemplateProps> = ({ data }) => {
 
-  return (<IndexLayout canonical={data.markdownRemark.frontmatter.url}>
+  return (<IndexLayout canonical={data.markdownRemark.frontmatter.url ? data.markdownRemark.frontmatter.url : `https://www.gitpod.io/blog/${data.markdownRemark.fields.slug}`}>
```
```suggestion
  return (<IndexLayout canonical={data.markdownRemark.frontmatter.url ? data.markdownRemark.frontmatter.url : `https://www.gitpod.io${data.markdownRemark.fields.slug}`}>
```

jankeromnes commented:
Also on a general note, even for small changes, it's much easier for both the developer and for the reviewer to make multiple smaller pull requests.

Bundling all changes into a single pull request makes it harder to review (takes more time, and the reviewer might overlook smaller problems because the PR is so big), and it's harder for the dev to iterate on multiple commits at the same time.

nisarhassan12 commented:
> Also on a general note, even for small changes, it's much easier for both the developer and for the reviewer to make multiple smaller pull requests.
> 
> Bundling all changes into a single pull request makes it harder to review (takes more time, and the reviewer might overlook smaller problems because the PR is so big), and it's harder for the dev to iterate on multiple commits at the same time.

Thanks! A lot I will try to keep this in mind from now on forward.

jankeromnes commented:
Also, I suggest you run:

```bash
git pull --rebase upstream master
```

In order to update your commits on top of the latest `master` branch.

nisarhassan12 commented:
> Also, I suggest you run:
> 
> ```shell
> git pull --rebase upstream master
> ```
> 
> In order to update your commits on top of the latest `master` branch.

Done Sir

jankeromnes commented:
Notes:
- I've rebased all commits on top of `upstream/master` again
- I've squashed all WIP commits into a single commit
- I've split this single commit into two parts again: the fixes I approve, and the fixes I'm not quite sure about yet

This will allow me to merge the first part (after making sure I didn't do any mistakes), and continue to review the second part.

In the future, please try to open multiple small pull requests, instead of making one giant pull request. I think this will make reviews & iterations & merges much easier. 😄

nisarhassan12 commented:
> Notes:
> 
>     * I've rebased all commits on top of `upstream/master` again
> 
>     * I've squashed all WIP commits into a single commit
> 
>     * I've split this single commit into two parts again: the fixes I approve, and the fixes I'm not quite sure about yet
> 
> 
> This will allow me to merge the first part (after making sure I didn't do any mistakes), and continue to review the second part.
> 
> In the future, please try to open multiple small pull requests, instead of making one giant pull request. I think this will make reviews & iterations & merges much easier. smile

Thanks! a lot @jankeromnes I will try to do that for sure. :pray:

jankeromnes commented:
Ok, I merged the first part of the changes. Thanks a lot Nisar! The canonical URLs especially seem all good now. 💯 

I'll try to post a review for the second part (commit `Small fixes 2/2`, which isn't merged yet) soon.
